src/main/java/org/jabref/logic/texparser/DefaultTexParser.java

```diff
@@ -44,69 +45,77 @@
 
     private static final String TEX_EXT = ".tex";
 
-    private final TexParserResult result;
+    private final TexParserResult texParserResult;
 
     public DefaultTexParser() {
-        this.result = new TexParserResult();
+        this.texParserResult = new TexParserResult();
     }
 
-    public TexParserResult getResult() {
-        return result;
+    public TexParserResult getTexParserResult() {
+        return texParserResult;
     }
 
     @Override
     public TexParserResult parse(String citeString) {
-        matchCitation(Paths.get("foo/bar"), 1, citeString);
-        return result;
+        matchCitation(null, Paths.get(""), 1, citeString);
+        return texParserResult;
     }
 
     @Override
     public TexParserResult parse(Path texFile) {
-        return parse(Collections.singletonList(texFile));
+        return parse(null, Collections.singletonList(texFile));
     }
 
     @Override
     public TexParserResult parse(List<Path> texFiles) {
-        List<Path> referencedFiles = new ArrayList<>();
+        return parse(null, texFiles);
+    }
+
+    /**
+     * Parse a list of TEX files for searching a given entry.
+     *
+     * @param entryKey String that contains the entry key we are searching (null for all entries)
+     * @param texFiles List of Path objects linked to a TEX file
+     * @return a TexParserResult, which contains all data related to the bibliographic entries
+     */
+    public TexParserResult parse(String entryKey, List<Path> texFiles) {
+        texParserResult.addFiles(texFiles);
 
-        result.addFiles(texFiles);
+        List<Path> referencedFiles = new ArrayList<>();
 
         for (Path file : texFiles) {
-            try (LineNumberReader lnr = new LineNumberReader(Files.newBufferedReader(file))) {
-                for (String line = lnr.readLine(); line != null; line = lnr.readLine()) {
-                    if (line.isEmpty() || line.charAt(0) == '%') {
-                        // Skip comments and blank lines.
-                        continue;
+            try (LineNumberReader lineNumberReader = new LineNumberReader(Files.newBufferedReader(file))) {
+                // Skip comments and blank lines.
+                lineNumberReader.lines().filter(line -> !line.isEmpty() && line.charAt(0) != '%').forEach(line -> {
+                    // Check if the current line contains a given entry (or 'entry' parameter is null).
+                    if (entryKey == null || line.contains(entryKey)) {
```

Comment:
Add a short comment explaining that the if check is there to speed-up the parsing.

Reply:
Ok. By the way, there is also a second filter in the `matchCitation()` method.

Reply:
There is no `Outdated` signal here because it is a comment, but now it says:

`// Skip the citation matching if the line does not contain the given entry to speed up the parsing.`